### DISCUSSION TITLE
passing parameters and secrets to powershell script

### DISCUSSION THREAD

**Discussion by lafayetteduarte** [semaphoreui/semaphore#2279] [Q&A]
Hi everyone.
I´m trying to figure out how to pass parameters to my powershell scripts and , so far came up empty.
Tried to:

1) Define a parameter on my ps1 file and create a survey variable . no luck
2) defined a variable in my environment and tried to list all defined variables inside my ps1  got nothing.

can anyone point me at an example or doc that shows how can i do that?

I am able to get the values from env variables but got no luck with any other method be that survey vars or simple vars in my environment settings.

My test script looks like this:

param(
    [string]$param1
)

$info = ""

$info = [System.Net.Dns]::GetHostName()

Write-Host $info

Write-Host "################################"
Write-Host "Parameters and Default Values:"
Write-Host "################################"
foreach ($param in $MyInvocation.MyCommand.Parameters.keys) {
    $value = Get-Variable -Name $param -ValueOnly -ErrorAction SilentlyContinue
    if (-not $null -eq [string]$value) {
        Write-Host "${param}: $value"
    }
}
Write-Host "################################"

Write-Host ""
Write-Host "################################"
Write-Host "Loaded variables:"
Write-Host "################################"
Get-Variable | Out-String
Write-Host "################################"
Write-Host ""

Write-Host ""
Write-Host "################################"
Write-Host "env variables:"
Write-Host "################################"
Get-ChildItem env:* | sort-object name
Write-Host "################################"
Write-Host ""

Thank you in advance

**Comment by acaronmd** [comment]
I ran your script to troubleshoot the same issue, and I think there's a bug.

Semaphore returns
`param1                         param1=param1value`

Whereas running the same script locally gives me
`param1                         param1value`

ENVs work correctly, but cannot be used to prompt the user for values during runtime.
